Task7/vertex-ai-qwikstart/utils/data_download.py

```diff
@@ -0,0 +1,188 @@
+            except:
+                logging.error('Error while deleting local tmp data directory.')
+
+        #print error if file doesn't exist.
+        except BaseException as error:
+            logging.error('An exception occurred: {}'.format(error))
+            
+    #print error if file already exists in GCS.
+    else:
+        logging.warning('File already exists in GCS.')
+            
+
+def upload_gcs2bq(args, schema):
+    """
+    args:
+    schema:
+    """    
+    # Construct a BigQuery client object.
+    client = bigquery.Client()
+    
+    # Construct a full Dataset object to send to the API.
+    logging.info('Initializing BigQuery dataset.')
+    dataset = bigquery.Dataset(f"{args.PROJECT_ID}.{args.BQ_DATASET_NAME}")
+
+    try:
+        # Send the dataset to the API for creation, with an explicit timeout.
+        # Raises google.api_core.exceptions.Conflict if the Dataset already
+        # exists within the project.
+        dataset = client.create_dataset(dataset, timeout=30)  # Make an API request.
+        # Specify the geographic location where the dataset should reside.
+        dataset.location = args.BQ_LOCATION
+    except Conflict:
+        logging.warning('Dataset %s already exists, not creating.', dataset.dataset_id)
+    else:
+        logging.info("Created dataset %s.%s", client.project, dataset.dataset_id)           
+    
+    try:
+        URI = f"gs://{args.GCS_BUCKET}/data/{FILENAME}.csv"
+        RAW_TABLE_ID = f"{args.PROJECT_ID}.{args.BQ_DATASET_NAME}.{args.BQ_RAW_TABLE_NAME}"
+        
+        # Load job.
+        job_config = bigquery.LoadJobConfig(
+            schema=schema,
+            skip_leading_rows=1,
+            allow_jagged_rows=True,
+            write_disposition="WRITE_TRUNCATE",
+            source_format=bigquery.SourceFormat.CSV)        
+        load_job = client.load_table_from_uri(source_uris=URI, destination=RAW_TABLE_ID, job_config=job_config)
+        logging.info('BQ raw dataset load job starting...')        
+        load_job.result()  # Waits for the job to complete.
+        logging.info('BQ raw dataset load job complete.')
+    except BaseException as error:
+        logging.error('An exception occurred: {}'.format(error))
+    
+    destination_table = client.get_table(RAW_TABLE_ID)  # Make an API request.
+    logging.info("Loaded %s rows into %s.",destination_table.num_rows, RAW_TABLE_ID)
+    
+
+def make_dataset_clean_bq(args, query: str):
+    """
+    args:
+    query:
+    """
+    client = bigquery.Client()
+    CLEAN_TABLE_ID = f"{args.PROJECT_ID}.{args.BQ_DATASET_NAME}.{args.BQ_CLEAN_TABLE_NAME}"
+    RAW_TABLE_ID = f"{args.PROJECT_ID}.{args.BQ_DATASET_NAME}.{args.BQ_RAW_TABLE_NAME}"    
+    
+    clean_query = query.replace("@CLEAN_TABLE_ID", CLEAN_TABLE_ID).replace("@RAW_TABLE_ID", RAW_TABLE_ID)
+
+    logging.info('BQ make clean dataset starting...')
+    try:
+      job = client.query(clean_query)
+      _ = job.result()
```

Comment:
Why return?

Reply:
Minor thin: Why not just job.result()? No need to return to a single _ ?

Reply:
Oh, I see. I will change it